mysql-test/suite/mysql_migration/mysql_migration.test

```diff
@@ -0,0 +1,206 @@
+--source include/restart_mysqld.inc 
+--exec $MYSQL_UPGRADE --force 2>&1
+
+--echo #
+--echo # Test that MariaDB can access tables partitioned with "native" InnoDb partition handler
+--echo #
+
+#
+# MariDB doc:
+#   MariaDB 10.3 does not support MySQL 5.7's “native” InnoDB partitioning handler.
+#
+# Please make sure the coresponding data exists in MySQL datadir via $POPULATE_MYSQL_DATADIR script
+#
+
+use idb_partition_db1;
+
+SHOW TABLES;
+
+--error ER_NO_SUCH_TABLE_IN_ENGINE # Allow the test to continue. Remove once this incompatibility is resolved
+select * from idb_partition_t1;
+
+# Confirm that MariaDB can still apply partitioning to the table as usual
+--error ER_NO_SUCH_TABLE_IN_ENGINE # Allow the test to continue. Remove once this incompatibility is resolved
+ALTER TABLE idb_partition_t1
+    PARTITION BY HASH(id)
+    PARTITIONS 8;
+
+--error ER_NO_SUCH_TABLE_IN_ENGINE # Allow the test to continue. Remove once this incompatibility is resolved
+select * from idb_partition_t1;
+
+--echo #
+--echo # Test that MariaDB can access tables compressed using MySQL punch-hole compression
+--echo #
+
+#
+# Please make sure the corresponding data exists in MySQL datadir via $POPULATE_MYSQL_DATADIR script
+#
+
+use idb_page_comp_db1;
+
+SHOW TABLES;
+
+--error ER_TABLE_CORRUPT # Allow the test to continue. Remove once this incompatibility is resolved
+select count(*) as EXPECT_3000 from idb_page_comp_t1;
+
+
+--echo #
+--echo # Test that MariaDB can access tables encrypted in MySQL with keyring_file plug-in
+--echo #
+
+#
+# MariDB doc:
+#   MariaDB 10.3 implements InnoDB encryption in a different way to MySQL 5.7
+#
+# Please make sure the corresponding data exists in MySQL datadir via $POPULATE_MYSQL_DATADIR script
+#
+
+use idb_encryption_db1;
+
+SHOW TABLES;
+
+--error ER_NO_SUCH_TABLE_IN_ENGINE # Allow the test to continue. Remove once this incompatibility is resolved
+select * from idb_encryption_t1;
+
+
+--echo #
+--echo # Test that table in tablespace created by MySQL CREATE TABLESPACE command is still accessible
+--echo #
+
+#
+# There is no plan to support CREATE TABLESPACE in near future, but at least mariadb-upgrade should
+# detect whether there is any custom table space in use, and either:
+#   1. Remove them so SHOW TABLES will not see them
+#   2. Convert them to File-Per-Table tablespaces so that it's fully accessible
+# This test assumes the second approach has been taken
+#
+# Please make sure the corresponding data exists in MySQL datadir via $POPULATE_MYSQL_DATADIR script
+#
+
+use idb_tablespace_db1;
+
+SHOW TABLES;
+
+--error ER_NO_SUCH_TABLE_IN_ENGINE # Allow the test to continue. Remove once this incompatibility is resolved
+select * from idb_tablespace_t1;
+
+
+--echo #
+--echo # Test that undo logs in MariaDB still work
+--echo #
+
+#
+# Because undo logs are implemented differently in MariDB and MySQL. This test is necessary to verify
+# that MariaDB undo logs works well with MySQL datadir
```

Comment:
The problem is that MySQL 5.7 introduced an incompatibility with MySQL 5.6, which MariaDB avoided, by applying 124bae082bf17e9af1fc77f78bbebd019635be5c.

If something was upgraded from MySQL 5.6 or earlier (`innodb_rollback_segments=128` was the default already in 5.5) to 5.7 and then to MariaDB, then the rollback segments 1 to 32 could contain garbage and cause some trouble. I do not think that a test of creating a database in 5.7 and upgrading to MariaDB can reveal much trouble with regard to that.

In any case, it could be useful to cause MySQL 5.7 rollback segments 1 to 32 to be written by executing operations on tables created with `CREATE TEMPORARY TABLE`.